### ISSUE TITLE
MacOS-Unable to launch Wesnoth 1.14.0-libreadline.dylib invalid code signature

### ISSUE THREAD

**Issue by audionuma** [wesnoth/wesnoth#3063]
Hello,

MacBook Pro 13" 2011 running MacOS 10.10.5 (Yosemite).

Running in an admin account.

Trying to run the provided Wesnoth 1.14.0 build downloaded from Sourceforge.

- Expend the downloaded dmg and copy the Wesnoth app to /Applications folder
- Right-click > Open on the app to launch it without security issues
- A MacOS error message is displayed (Wesnot cannot be open because of a problem ... ) with the option to report.

Selecting report displays a report :

```
Process:               Wesnoth [3015]
Path:                  /Applications/Wesnoth.app/Contents/MacOS/Wesnoth
Identifier:            org.wesnoth.Wesnoth
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Wesnoth [3015]
User ID:               501

Date/Time:             2018-05-09 14:01:56.306 +0200
OS Version:            Mac OS X 10.10.5 (14F2511)
Report Version:        11
Anonymous UUID:        C0BA15CE-9D84-9C1D-9E5A-53F50F17BD7A

Sleep/Wake UUID:       B975E605-EF80-4309-BC3E-2EDF3DE8181A

Time Awake Since Boot: 50000 seconds
Time Since Wake:       1000 seconds

Crashed Thread:        0

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000002, 0x0000000000000000

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @executable_path/../Frameworks/libreadline.7.0.dylib
  Referenced from: /Applications/Wesnoth.app/Contents/MacOS/Wesnoth
  Reason: no suitable image found.  Did find:
	/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib: code signature invalid for '/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib'

	/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib: code signature invalid for '/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib'

Binary Images:
    0x7fff60b80000 -     0x7fff60bb692f  dyld (353.2.4) <3C068A0D-69D5-3AD6-86D9-6261D3B1564E> /usr/lib/dyld
    0x7fff83e5a000 -     0x7fff83e6bff7  libz.1.dylib (55) <88C7C7DE-04B8-316F-8B74-ACD9F3DE1AA1> /usr/lib/libz.1.dylib
    0x7fff8aa51000 -     0x7fff8aa5eff7  libbz2.1.0.dylib (36) <2DF83FBC-5C08-39E1-94F5-C28653791B5F> /usr/lib/libbz2.1.0.dylib

Model: MacBookPro8,1, BootROM MBP81.0047.B2D, 2 processors, Intel Core i7, 2.7 GHz, 4 GB, SMC 1.68f99
Graphics: Intel HD Graphics 3000, Intel HD Graphics 3000, Built-In
Memory Module: BANK 0/DIMM0, 2 GB, DDR3, 1333 MHz, 0x80CE, 0x4D34373142353737334448302D4348392020
Memory Module: BANK 1/DIMM0, 2 GB, DDR3, 1333 MHz, 0x80CE, 0x4D34373142353737334448302D4348392020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0xD6), Broadcom BCM43xx 1.0 (5.106.98.100.24)
Bluetooth: Version 4.3.6f4 17916, 3 services, 18 devices, 1 incoming serial ports
Network Service: Ethernet, Ethernet, en0
Serial ATA Device: APPLE SSD TS256C, 251 GB
Serial ATA Device: OPTIARC DVD RW AD-5970H
USB Device: FaceTime HD Camera (Built-in)
USB Device: Hub
USB Device: Apple Internal Keyboard / Trackpad
USB Device: BRCM2070 Hub
USB Device: Bluetooth USB Host Controller
USB Device: Hub
USB Device: USB Optical Mouse
USB Device: IR Receiver
Thunderbolt Bus: MacBook Pro, Apple Inc., 22.1

```
There seems to be an issue with the signature of the libreadline.7.0.dylib.

For info, my security settings are set on _Allow downloaded applications from MacApp store and identified developers_.

Changing the setting to _Anywhere_ doesn't solve the issue.

**Comment by audionuma** [comment]
Thanks for the updated package. Testing macOS package Wesnoth 1.14.1 : still having the (same ?) issue.

```
Process:               Wesnoth [3503]
Path:                  /Applications/Wesnoth.app/Contents/MacOS/Wesnoth
Identifier:            org.wesnoth.Wesnoth
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Wesnoth [3503]
User ID:               501

Date/Time:             2018-05-09 16:07:50.899 +0200
OS Version:            Mac OS X 10.10.5 (14F2511)
Report Version:        11
Anonymous UUID:        C0BA15CE-9D84-9C1D-9E5A-53F50F17BD7A

Sleep/Wake UUID:       B975E605-EF80-4309-BC3E-2EDF3DE8181A

Time Awake Since Boot: 58000 seconds
Time Since Wake:       8500 seconds

Crashed Thread:        0

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000002, 0x0000000000000000

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @executable_path/../Frameworks/libreadline.7.0.dylib
  Referenced from: /Applications/Wesnoth.app/Contents/MacOS/Wesnoth
  Reason: no suitable image found.  Did find:
	/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib: code signature invalid for '/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib'

	/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib: code signature invalid for '/Applications/Wesnoth.app/Contents/MacOS/../Frameworks/libreadline.7.0.dylib'

Binary Images:
    0x7fff6c387000 -     0x7fff6c3bd92f  dyld (353.2.4) <3C068A0D-69D5-3AD6-86D9-6261D3B1564E> /usr/lib/dyld
    0x7fff83e5a000 -     0x7fff83e6bff7  libz.1.dylib (55) <88C7C7DE-04B8-316F-8B74-ACD9F3DE1AA1> /usr/lib/libz.1.dylib
    0x7fff8aa51000 -     0x7fff8aa5eff7  libbz2.1.0.dylib (36) <2DF83FBC-5C08-39E1-94F5-C28653791B5F> /usr/lib/libbz2.1.0.dylib

Model: MacBookPro8,1, BootROM MBP81.0047.B2D, 2 processors, Intel Core i7, 2.7 GHz, 4 GB, SMC 1.68f99
Graphics: Intel HD Graphics 3000, Intel HD Graphics 3000, Built-In
Memory Module: BANK 0/DIMM0, 2 GB, DDR3, 1333 MHz, 0x80CE, 0x4D34373142353737334448302D4348392020
Memory Module: BANK 1/DIMM0, 2 GB, DDR3, 1333 MHz, 0x80CE, 0x4D34373142353737334448302D4348392020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0xD6), Broadcom BCM43xx 1.0 (5.106.98.100.24)
Bluetooth: Version 4.3.6f4 17916, 3 services, 18 devices, 1 incoming serial ports
Network Service: Ethernet, Ethernet, en0
Serial ATA Device: APPLE SSD TS256C, 251 GB
Serial ATA Device: OPTIARC DVD RW AD-5970H
USB Device: FaceTime HD Camera (Built-in)
USB Device: Hub
USB Device: Apple Internal Keyboard / Trackpad
USB Device: BRCM2070 Hub
USB Device: Bluetooth USB Host Controller
USB Device: Hub
USB Device: USB Optical Mouse
USB Device: IR Receiver
Thunderbolt Bus: MacBook Pro, Apple Inc., 22.1

```

**Comment by hrubymar10** [comment]
Ok, it is possibly this issue: https://stackoverflow.com/questions/41865537/how-does-apples-codesign-utility-decide-which-sha-algorithms-to-sign-a-shared/41868747#41868747

I'll update libraries inside macOS package and create new package. Please contact me over Discord `hrubymar10#0681` or Email `hrubymar10 at gmail dot com`. I'll send you link to new package and if it will work for you I'll release it for all.

**Comment by audionuma** [comment]
You can send me the link to audionuma@gmail.com
Thanks.

**Comment by hrubymar10** [comment]
I fixed all problems in re-released macOS package `Wesnoth-1.14.1a.dmg`

Available here: https://sourceforge.net/projects/wesnoth/files/wesnoth-1.14/wesnoth-1.14.1/

Closing
